src/pip/_internal/operations/install/wheel.py

```diff
@@ -583,6 +576,51 @@ def is_entrypoint_wrapper(name):
                 filter=filter,
             )
 
+    def pyc_source_file_paths():
+        # type: () -> Iterator[text_type]
+        # We de-duplicate installation paths, since there can be overlap (e.g.
+        # file in .data maps to same location as file in wheel root).
+        # Sorting installation paths makes it easier to reproduce and debug
+        # issues related to permissions on existing files.
+        for installed_path in sorted(set(installed.values())):
+            full_installed_path = os.path.join(lib_dir, installed_path)
+            if not os.path.isfile(full_installed_path):
+                continue
+            if not full_installed_path.endswith('.py'):
+                continue
+            yield full_installed_path
+
+    def pyc_output_path(path):
```

Comment:
This might be better moved to utils, as it's basically a compatibility wrapper around `importlib.util.cache_from_source`. Once we drop Python 2 support, we should remove this in favour of calling importlib directly.

Reply:
If/when we switch to `py_compile.compile` I expect this to go away (or only be used inside it) since we'd need a compatibility wrapper for that as well.